django/middleware/csrf.py

```diff
@@ -221,7 +222,41 @@ def process_view(self, request, callback, callback_args, callback_kwargs):
                 # branches that call reject().
                 return self._accept(request)
 
-            if request.is_secure():
+            # Verify the Origin header, if present.
+            origin_verified = False
+            if 'HTTP_ORIGIN' in request.META:
+                good_origin = '%s://%s' % (
```

Comment:
In comparision to the `Refer(r)er` check we loose the possibility to override the "initial" override of `get_host`. Would this be something to worry about -- not sure, just thinking out loud…

Reply:
I think you're pointing out that `settings.SESSION_COOKIE_DOMAIN` (and `settings.CSRF_COOKIE_DOMAIN`) aren't considered in origin checking like they are in referer checking? I think it's simpler to require use of `CSRF_TRUSTED_ORIGINS` (as the release note suggests may be required) rather than migrating those settings to include a scheme so they could be used in origin checking.

Reply:
I agree that it is simpler. But now the pre-calculated `good_origin` is always part of the allowed origins. I cannot see why this would be a problem, I just wanted to put it out so we can think about it :)